pkg/ctl/cmdutils/cmdutils.go

```diff
@@ -124,9 +124,28 @@ func AddTimeoutFlag(fs *pflag.FlagSet, p *time.Duration) {
 	AddTimeoutFlagWithValue(fs, p, api.DefaultWaitTimeout)
 }
 
-// AddNameFlag adds common --name flag for cluster
-func AddNameFlag(fs *pflag.FlagSet, meta *api.ClusterMeta) {
+// AddClusterFlag adds a common --cluster flag for cluster name.
+// Use this for commands whose principal resource is *not* a cluster.
+func AddClusterFlag(fs *pflag.FlagSet, meta *api.ClusterMeta) {
+	fs.StringVarP(&meta.Name, "cluster", "c", "", "EKS cluster name")
+}
+
+// AddClusterFlagWithDeprecated adds a common --cluster flag for
+// cluster name as well as a deprecated --name flag.
+// Use AddClusterFlag() for new commands.
+func AddClusterFlagWithDeprecated(fs *pflag.FlagSet, meta *api.ClusterMeta) {
+	AddClusterFlag(fs, meta)
 	fs.StringVarP(&meta.Name, "name", "n", "", "EKS cluster name")
+	_ = fs.MarkDeprecated("name", "use --cluster")
```

Comment:
How about we use `MakeHidden` and produce a warning when it is used to begin with. After 3 release we can switch to `MarkDeprecated`?

Reply:
> How about we use MakeHidden and produce a warning when it is used to begin with

my understanding is this is exactly what `MarkDeprecated` does

Reply:
it doesn't exit, see https://github.com/weaveworks/eksctl/pull/1363#issuecomment-537191382

our logger doesn't support writing to `stderr` (unless we enable something called `TestMode` that turns off colored output) which `MarkDeprecated` does and I would consider an important part to make sure not to break things for clients that parse the output for now.

also if we re-implement this ourselves it needs a fair amount of extra code and new people joining the project need to discover this. `MarkDeprecated` is known by everyone that knows cobra.

> it would be good make the warning more informative, to say that flag will be removed in a specific release.

can do this in a new PR, we have a couple deprecated flags. also not pressing to remove those IMHO.

Reply:
Mark deprecated just prints a warning but it allows the operation to continue. I just tested this. Here is an example:

```
$ eksctl create iamidentitymapping --name martina-test-4 --role arn:aws:iam::123:user/eksctl-testing-XYZ --group "system:something"
Flag --name has been deprecated, use --cluster
Flag --role has been deprecated, use --arn
[ℹ]  eksctl version 
[ℹ]  using region eu-north-1
[ℹ]  adding identity "arn:aws:iam::123:user/eksctl-testing-XYZ" to auth ConfigMap

$ eksctl create iamidentitymapping --cluster  martina-test-4 --role arn:aws:iam::123:user/eksctl-testing-XYZ --group "system:something"
Flag --role has been deprecated, use --arn
[ℹ]  eksctl version 
[ℹ]  using region eu-north-1
[!]  found existing mappings with same arn "arn:aws:iam::123:user/eksctl-testing-XYZ" (which will be shadowed by your new mapping)
[ℹ]  adding identity "arn:aws:iam::123:user/eksctl-testing-XYZ" to auth ConfigMap

```

I think with that we can proceed with the PR so I will merge it.